File: algo-deaf-grandma/deaf_grandma.js
// we need a way to get user input
// we need to analyze the input and react appropriately
// - say nothing
// - lower case
// - goodbye (first and second)
// - normal question (all caps)


function deafGrandma() {

    let goodbyeCounter = 0
    let input = ""
    let response = ""
    
    // grandma starts conversation
    console.log("HEY, KID!")
    while (goodbyeCounter <= 1) { // continue talking to grandma so long as we haven't said goodbye more than once
      input = window.prompt("Respond to grandma:", "hello")
      // - say nothing
      if (input === "") {
        response = "WHAT?!"
      // - lower case
      } else if (input !== input.toUpperCase()) {
        response = "SPEAK UP, KID!"
      // - normal question (all caps)
      } else if (input === "GOODBYE!") {
        goodbyeCounter = goodbyeCounter + 1
        if (goodbyeCounter > 1) { // we've said goodbye once before
          response = "LATER, SKATER!"
        } else { // this is the first time we're saying goodbye
          response = "LEAVING SO SOON?"
        }
      } else {
        response = "NO, NOT SINCE 1946!"
      }
      console.log(response)
    }
  }
  
  deafGrandma();

File: algo-deaf-grandma/deaf_grandma.py
def deaf_grandma():
        import re
        goodbye = 0

        print('HEY KID!')
        while goodbye < 2:
                says = input()
                if says == "":
                        print("WHAT?!")
                elif says == "GOODBYE!" and goodbye == 0:
                    goodbye += 1
                    print("LEAVING SO SOON?")
                elif says == "GOODBYE!" and goodbye == 1:
                    goodbye = 2
                    print("LATER SKATER!")
                elif re.search("[a-z]",says) != None:
                    print("SPEAK UP, KID!")
                elif re.search("[A-Z]+",says) != None:
                    print("NO, NOT SINCE 1946!")
                elif re.search("[a-zA-Z]+", says) is None:
                        print("SOMETHIN WRONG WITH YOU? YOURE MOVIN YOUR MOUTH BUT NOTHING IS COMING OUT. SAY IT WITH YOUR CHEST SONNY!")


deaf_grandma()
